libs/ledger-live-common/src/families/cosmos/api/Cosmos.ts

```diff
@@ -83,16 +92,30 @@ export class CosmosAPI {
         url: `${this.defaultEndpoint}/cosmos/auth/${this.version}/accounts/${address}`,
       });
 
-      if (data.account.account_number) {
-        response.accountNumber = parseInt(data.account.account_number);
+      // We use base_account for Ethermint chains and account for the rest
+      const srcAccount = data.account.base_account || data.account;
+
+      if (srcAccount.account_number) {
+        accountData.accountNumber = parseInt(srcAccount.account_number);
+      }
+
+      if (srcAccount.sequence) {
+        accountData.sequence = parseInt(srcAccount.sequence);
       }
 
-      if (data.account.sequence) {
-        response.sequence = parseInt(data.account.sequence);
+      if (srcAccount.pub_key) {
```

Comment:
Just curious, I don't remember if I asked already: did you see cases where the `pub_key` field isn't provided?
If it wasn't optional, then we wouldn't need `defaultPubKeyType` at all, but I assume it is optional?

Reply:
pub_key is null when account is pristine, we want to use default value else we end up on the bug i resolved last week